### PR TITLE
[INTEL-HPU] Deepseek R1 model enabling for Intel Gaudi

### PR DESCRIPTION
**Pull request by xuechendi**
updated in 03/24/2025

convert this PR as a draft and waiting for 1.21 HPU synapse release

----

Accuracy is verified as below
```
VLLM_SKIP_WARMUP=true \
PT_HPU_ENABLE_LAZY_COLLECTIVES=true \
PT_HPU_WEIGHT_SHARING=0 \
VLLM_TEST_ENABLE_EP=1 \
VLLM_MLA_DISABLE_REQUANTIZATION=1 \
bash .buildkite/lm-eval-harness/run-lm-eval-gsm-vllm-baseline.sh -m deepseek-ai/DeepSeek-R1 -b 128 -l 256 -f 5 -t 8
```
image

3 options for running fp8 on HPU:
1. dynamic quant + static MOE => **default option** . Supported by public released hpu 1.20 synapse container
2. dynamic quant + dynamic MOE => enabled by "VLLM_DMOE_DYNAMIC_SCALE=1". Working with internal hpu synapse build and will release container later
3. static quant + dynamic MOE => enabled by using a converted model from original block-based quant to channel-based quant. Please refer https://github.com/HabanaAI/vllm-fork/blob/deepseek_r1/scripts/DEEPSEEK_R1_ON_GAUDI.md#option-3-run-with-static-quantization for instruction of how to convert model

### CODE REVIEW

File: vllm/attention/backends/hpu_attn.py [hunk 1]
```diff
@@ -8,20 +8,90 @@
...
+import habana_frameworks.torch as htorch  # noqa: F401
 import torch
 import vllm_hpu_extension.ops as ops
+from neural_compressor.torch.algorithms.fp8_quant._core.quant_dequant import (
+    DequantOutput, QuantInput)
+from neural_compressor.torch.algorithms.fp8_quant._quant_common.helper_modules import (  # noqa: E501
+    PatchedVLLMKVCache)
+from neural_compressor.torch.algorithms.fp8_quant._quant_common.quant_config import (  # noqa: E501
+    Fp8cfg)
+from neural_compressor.torch.algorithms.fp8_quant.model_configs import (
```
Comment (yangw1234) [inline vllm/attention/backends/hpu_attn.py:20]: should we add this to dependencies?

Comment (xuechendi) [reply vllm/attention/backends/hpu_attn.py:20]: > should we add this to dependencies?

Let me check with INC team.

**Comment by xuechendi** [comment]
@mgoin @robertgshaw2-redhat @WoosukKwon @simon-mo @youkaichao @comaniac 
Would like to get some reviews, thanks
